T4MVCExtensions/T4Extensions.cs

```diff
@@ -334,9 +334,11 @@ public static string JavaScriptReplaceableUrl(this UrlHelper urlHelper, ActionRe
             string action = token.ToString();
 
             var specificActionUrl = (from r in RouteTable.Routes.OfType<Route>()
-                                     where r.DataTokens != null && r.Defaults != null
-                                     && r.DataTokens.Any(dt => string.Compare(dt.Key, "area", StringComparison.InvariantCultureIgnoreCase) == 0
-                                         && string.Compare(dt.Value.ToString(), area, StringComparison.InvariantCultureIgnoreCase) == 0)
+                                     where r.Defaults != null
+                                     && ((string.IsNullOrEmpty(area) 
+                                        && (r.DataTokens == null || !r.DataTokens.Any(dt => string.Compare(dt.Key, "area", StringComparison.InvariantCultureIgnoreCase) == 0)))
+                                        || (r.DataTokens != null && r.DataTokens.Any(dt => string.Compare(dt.Key, "area", StringComparison.InvariantCultureIgnoreCase) == 0 
+                                            && string.Compare(dt.Value.ToString(), area, StringComparison.InvariantCultureIgnoreCase) == 0)))
```

Comment:
Cool. Not blaming you for the messy code, as it grew that way over time as things usually do! If you have time to do a little refactoring, that would be welcome!